aesara/graph/opt.py

```diff
@@ -1774,26 +1770,30 @@ def retry_with_equiv():
             return u
 
         u = match(self.in_pattern, node.out, unify.Unification(), True, self.pdb)
-        if u:
-
-            def build(pattern, u):
-                if isinstance(pattern, (list, tuple)):
-                    args = [build(p, u) for p in pattern[1:]]
-                    return pattern[0](*args)
-                elif isinstance(pattern, str):
-                    return u[unify.Var(pattern)]
-                elif isinstance(pattern, (int, float)):
-                    return pattern
-                else:
-                    return pattern.clone()
+        if not u:
+            return False
+
+        def build(pattern, u):
+            if isinstance(pattern, (list, tuple)):
+                args = [build(p, u) for p in pattern[1:]]
+                return pattern[0](*args)
+            elif isinstance(pattern, str):
+                return u[unify.Var(pattern)]
+            elif isinstance(pattern, (int, float)):
+                return pattern
+            else:
+                return pattern.clone()
 
-            p = self.out_pattern
-            ret = build(p, u)
-            if self.values_eq_approx:
-                ret.tag.values_eq_approx = self.values_eq_approx
+        ret = build(self.out_pattern, u)
+
+        if isinstance(ret, (int, float)):
+            # TODO: Should we convert these to constants explicitly?
```

Comment:
Remember, **all this is going to be replaced with a more capable/mature unification library**, so there aren't really any TODOs here.